### PR TITLE
Adding method to parse the payload of an asset transfer

### PR DESCRIPTION
**Pull request by sallymoc**

### CODE REVIEW

File: src/qubic-types/transacion-payloads/QubicTransferAssetPayload.ts [hunk 1]
```diff
@@ -101,4 +101,46 @@ export class QubicTransferAssetPayload implements IQubicBuildPackage {
...
+
+        const amount = Number(
+            this.uint8ArrayToBigInt(data.slice(start, end)) as any
+        );
+        this.numberOfUnits = new Long(amount);
+
+        return this;
+    }
+
+    uint8ArrayToBigInt(bytes: Uint8Array): bigint {
```
[the remark this answers is not in the document]
Comment (J0ET0M) [reply src/qubic-types/transacion-payloads/QubicTransferAssetPayload.ts:140]: could extend the "Long" class to accept uint8array

**Review by J0ET0M — COMMENTED** [review]
looks good. can you please adjust the one commented.

File: src/qubic-types/transacion-payloads/QubicTransferAssetPayload.ts [hunk 2]
```diff
@@ -101,4 +101,37 @@ export class QubicTransferAssetPayload implements IQubicBuildPackage {
...
+
+        start = end;
+        end = start + 8; // size for asset name
+        this.assetName = data.slice(start, end);
+
+        let decoder = new TextDecoder(); // Create a TextDecoder for UTF-8 by default
+        const result = decoder.decode(this.assetName); // Convert Uint8Array to string
+
+        start = end;
+        end = start + 8; // size for asset name
```
Comment (J0ET0M) [inline src/qubic-types/transacion-payloads/QubicTransferAssetPayload.ts:130]: i guess this should be size of number of units

File: src/qubic-types/transacion-payloads/QubicTransferAssetPayload.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sallymoc) [reply src/qubic-types/transacion-payloads/QubicTransferAssetPayload.ts]: fixed, please review

**Review by J0ET0M — APPROVED** [review]

**Pull request merged**